### ISSUE TITLE
vendor the puppet gem

### ISSUE THREAD

**Issue by gepandz** [logicminds/puppet-retrospec#17]
I *almost* upgraded my Puppet Enterprise today from Puppet 3.2 to Puppet 3.7.3 because of the Gemfile's line 3: `gem 'puppet', '= 3.7.3'`. We intend to upgrade from 3.2 this year, just not this afternoon. :)

Do you have a real dependency on a feature of 3.7.3, or could you support, say, ">= 3.2"? If you need the newer featureset, would you at least put a warning in the README that this might upgrade you to a higher version of Puppet? Thanks!

**Comment by logicminds** [comment]
I use RVM and have unique gemsets for every single puppet module so I don't really have this issue since retrospec is installed in my global gemset while my puppet module gems are installed in a local gemset for each puppet module.  You can use bundler to create unique gemsets for each module instead of relying on the global gemset which is where your puppet version conflict comes into play.  

```shell
bundle config path .bundle (only need to do this once)
cd puppet_module 
bundle install  (reads Gemfile and installs all gems inside .bundle)
```

Of course this will change your workflow a bit because now many things need to be run as `bundle exec rake spec, bundle exec rake ...`
I don't know if this is the proper way to write modules but it works really well for me and I never have to worry about puppet version conflicts.
I think that I can "vendor" the puppet gem inside my gem which would get rid of this issue entirely without changing someone's workflow.  But I would really recommend using the bundler gemset features.

Thanks for the feedback I'll definitely update my README and see if I can make retrospec bring its own puppet version.

**Comment by gepandz** [comment]
Thanks for the quick feedback! We'll take a look at bundling, and I appreciate your offer to vendor the Puppet gem and update the README to make it easier for others. Have a good one!

**Comment by logicminds** [comment]
I just ran into this bug as a result of not using bundler.  The puppet gem will be vendored in the next release. @gepandz

**Comment by logicminds** [comment]
fixed in 0.7.2
